tests/js/pv-tests.js

```diff
@@ -0,0 +1,218 @@
+        name: "string",
+        realm_id: "string?"
+    }
+};
+
+const PvUuidDog = {
+    name: "Dog",
+    primaryKey: "_id",
+    properties: {
+        _id: "objectId?", // NOTE: this needs to be changed to non-optional in the docker image.
+        breed: "string?",
+        name: "string",
+        realm_id: "uuid?"
+    }
+};
+
+const PvObjectidDog = {
+    name: "Dog",
+    primaryKey: "_id",
+    properties: {
+        _id: "objectId?", // NOTE: this needs to be changed to non-optional in the docker image.
+        breed: "string?",
+        name: "string",
+        realm_id: "objectId?"
+    }
+};
+
+module.exports = {
+    async testPvInt() {
+        let app = new Realm.App(AppConfig.pvIntAppConfig);
+
+        let credentials = Realm.Credentials.anonymous();
+        let user = await app.logIn(credentials);
+        const realmConfig = {
+            schema: [PvIntDog],
+            sync: {
+                user: user,
+                partitionValue: 42,
+                _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+            }
+        };
+
+        Realm.deleteFile(realmConfig);
+        let realm1 = await Realm.open(realmConfig);
+        realm1.write(() => {
+            realm1.create("Dog", { "_id": new ObjectId(), name: "King" });
+        });
+
+        await realm1.syncSession.uploadAllLocalChanges();
+        TestCase.assertEqual(realm1.objects("Dog").length, 1);
+        realm1.close();
+
+        Realm.deleteFile(realmConfig);
+
+        let realm2 = await Realm.open(realmConfig);
+        await realm2.syncSession.downloadAllServerChanges();
+
+        let dogs = realm2.objects("Dog");
+        TestCase.assertEqual(dogs.length, 1);
+
+        realm2.close();
+        await user.logOut();
+    },
+
+    async testPvString() {
+        let app = new Realm.App(AppConfig.pvStringAppConfig);
+
+        let credentials = Realm.Credentials.anonymous();
+        let user = await app.logIn(credentials);
+        const realmConfig = {
+            schema: [PvStringDog],
+            sync: {
+                user: user,
+                partitionValue: "42",
+                _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+            }
+        };
+
+        Realm.deleteFile(realmConfig);
+        let realm1 = await Realm.open(realmConfig);
+        realm1.write(() => {
+            realm1.create("Dog", { "_id": new ObjectId(), name: "King" });
+        });
+
+        await realm1.syncSession.uploadAllLocalChanges();
+        TestCase.assertEqual(realm1.objects("Dog").length, 1);
+        realm1.close();
+
+        Realm.deleteFile(realmConfig);
+
+        let realm2 = await Realm.open(realmConfig);
+        await realm2.syncSession.downloadAllServerChanges();
+
+        let dogs = realm2.objects("Dog");
+        TestCase.assertEqual(dogs.length, 1);
+
+        realm2.close();
+        await user.logOut();
+    },
+
+    /*async testUuidString() {
```

Comment:
[Question]  Are these commented-out tests waiting for something to be enabled in Stitch, or were they meant to be removed?

Reply:
Yes, these were awaiting a core update.